the-collab-lab/tcl-16-smart-shopping-list
# 7. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works. 🐪

## Description

First, this will create a new structure for ListView where a new parent component ListContainer will contain conditional rendering to prompt users who already have a token, but no items in their list to add items via the new component EmptyList. Users who have a token and items in their list will just see a regular ListView rendered.

We felt it was necessary to create conditional rendering here to differentiate from new users with an inherent empty list and existing users with an empty list of their own doing. We felt it was ultimately more efficient to preserve the excellent code in the existing ListView while creating a higher level check.

We learned more about accessing the database correctly, allowing for async returns, and deserializing objects.

## Related Issue

Closes #7 

## Acceptance Criteria

* The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
| ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

[![Image from Gyazo](https://gyazo.com/68e93fc980bb98b249042044af25d25c.gif)](https://gyazo.com/68e93fc980bb98b249042044af25d25c)

### After

[![Empty List](https://i.gyazo.com/924030919296c6361bdb799fefd1f412.gif)]


## Testing Steps / QA Criteria

1. remove token and check that new user functionality is in tact with new token generated
2. create a list full of items and then delete all items to check that EmptyList renders with "Add Item" button and redirects to AddItem component
